lib/ruby_lsp/document.rb

```diff
@@ -87,15 +87,15 @@ def push_edits(edits, version:)
       end
 
       @version = version
-      @unparsed_edits.concat(edits)
+      @needs_parsing = true
       @cache.clear
     end
 
     sig { void }
     def parse
-      return if @unparsed_edits.empty?
+      return unless @needs_parsing
 
-      @unparsed_edits.clear
+      @needs_parsing = false
```

Comment:
Instead of the instance variable, would it be feasible to set `@parse_result` to nil in `push_edits` and use that to determine whether we need to parse again?

Reply:
That would be risky since the LSP is multi-threaded. If we switch to a different thread that is executing another request and requires the parse result, then it would break.